errormessages/src/ts/main.tsx

```diff
@@ -0,0 +1,108 @@
+///**
+// * This Source Code Form is subject to the terms of the Mozilla Public
+// * License, v. 2.0. If a copy of the MPL was not distributed with this
+// * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+// */
+
+/// <reference path="../tsd/tsd.d.ts" />
+
+import * as cu from 'cu-core';
+import events from 'cu-events';
+import * as React from 'react/addons';
+
+var ReactCSSTransitionGroup = React.addons.CSSTransitionGroup;
+declare const cuAPI: any;
+
+class ErrorMessage extends React.Component<any, any> {
+
+    render() {
+       
+        var commentNodes = this.props.messageList.map(function (comment: string, index: number) {
+            return (
+                <li className='message' key={index + comment}>{comment}</li>
+            );
+        });
+
+        //TODO: figure out why <ReactCSSTransitionGroup transitionName="example"> isn't working
+        return (
+            <ul id='messages'>
+                {commentNodes}
+            </ul>
+        );
+    }
+}
+
+/* 
+*  Trying to set the state as a Array<string> was throwing errors so I
+*  wrapped the array in a class (like the KOS example) and it worked.
+*  Not sure why but I'll go with this for now
+*/
+class ErrorMessagesAppState {
+    public items: Array<string>;
+
+    constructor() {
+        this.items = new Array<string>();
+    }
+}
+
+class ErrorMessagesApp extends React.Component<any, ErrorMessagesAppState> {
+
+    constructor(props: any) {
+        super(props);
+
+        this.state = new ErrorMessagesAppState();
+
+        this.init = this.init.bind(this);
+        this.removeMessage = this.removeMessage.bind(this);
+        events.on('init', this.init);
+    }
+
+    init() {
+        //todo: event-ize OnAbilityError
+        cuAPI.OnAbilityError((message: any) => {
```

Comment:
Yes, for this I think a simple ability-error event in cu-events would be appropriate, I can imagine it is something a UI might want to hook into without being bothered with the rest of the ability stuff.

events.on('ability-error', ...)